Wikipedia/Code/WMFNotificationsController.m

```diff
@@ -27,51 +24,74 @@
 
 @interface WMFNotificationsController ()
 
-@property (nonatomic, readwrite, getter=isAuthorized) BOOL authorized;
 @property (weak, nonatomic) MWKDataStore *dataStore;
+@property (nonatomic, readwrite, copy, nullable) NSData *remoteRegistrationDeviceToken;
+@property (nonatomic, readwrite, strong, nullable) NSError *remoteRegistrationError;
+@property (nonatomic, strong) WMFEchoSubscriptionFetcher *echoSubscriptionFetcher;
+@property (nonatomic, strong) MWKLanguageLinkController *languageLinkController;
 
 @end
 
 @implementation WMFNotificationsController
 
-- (instancetype)initWithDataStore:(MWKDataStore *)dataStore {
+- (instancetype)initWithDataStore:(MWKDataStore *)dataStore languageLinkController:(MWKLanguageLinkController *)languageLinkController {
     self = [super init];
     if (self) {
         self.dataStore = dataStore;
-#if WMF_ALWAYS_ASK_FOR_NOTIFICATION_PERMISSION
-        [self requestAuthenticationIfNecessaryWithCompletionHandler:^(BOOL granted, NSError *_Nullable error){
-
-        }];
-#endif
+        self.languageLinkController = languageLinkController;
+        self.echoSubscriptionFetcher = [[WMFEchoSubscriptionFetcher alloc] initWithSession:dataStore.session configuration:dataStore.configuration];
     }
     return self;
 }
 
-- (void)requestAuthenticationIfNecessaryWithCompletionHandler:(void (^)(BOOL granted, NSError *__nullable error))completionHandler {
+- (void)notificationPermissionsStatusWithCompletionHandler:(void (^)(UNAuthorizationStatus  status))completionHandler {
+    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+    [center getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *_Nonnull settings) {
+        completionHandler(settings.authorizationStatus);
+    }];
+}
+
+- (void)requestPermissionsIfNecessaryWithCompletionHandler:(void (^)(BOOL isAllowed, NSError *__nullable error))completionHandler {
     if (![UNUserNotificationCenter class]) {
         completionHandler(NO, nil);
         return;
```

Comment:
I know this wasn't added in this PR, but I wonder if we can just remove this historical code (also in `sendNotificationWithTitle:`). Maybe I'm missing something, but perhaps it was just some pre-`UNUserNotificationCenter` safety code?

Reply:
I think that's a good suggestion - I pushed a commit removing these checks.